lib/truffle/truffle/cext.rb

```diff
@@ -1612,6 +1612,8 @@ def rb_rescue(b_proc, data1, r_proc, data2)
     begin
       Primitive.call_with_c_mutex(b_proc, [data1])
     rescue StandardError => e
+      # Allow rescues to be null, or else Graal will error.
+      return if Truffle::Interop.null?(r_proc)
```

Comment:
Rather than an explicit return, could you use a postfix `unless` here?
`return` in `ensure` seems kind of anti-pattern.